src/integrations/openExchangeRates.ts

```diff
@@ -0,0 +1,71 @@
+import { env } from 'node:process';
+import { ONE_DAY_IN_SECONDS } from '../common';
+import { logger } from '../logger';
+import { getRedisHashField, setRedisHashWithExpiry } from '../redis';
+import { isNullOrUndefined } from '../common/object';
+import { retryFetchParse } from './retry';
+import { StorageKey } from '../config';
+
+const REDIS_EXPIRATION = ONE_DAY_IN_SECONDS;
+
+const URL = 'https://openexchangerates.org/api/latest.json';
+
+export type CurrencyRate = Record<string, number>;
+
+export type OpenExchangeRates = {
+  disclaimer: string;
+  license: string;
+  timestamp: number;
+  base: string;
+  rates: CurrencyRate;
+};
+
+export const getExchangeRate = async (
+  currency: string,
+): Promise<number | null> => {
+  const rate = await getRedisHashField(StorageKey.OpenExchangeRates, currency);
+  if (isNullOrUndefined(rate)) {
+    return null;
+  }
+
+  return parseFloat(rate);
+};
+
+export const getOpenExchangeRates = async (): Promise<void> => {
+  if (!env.OPEN_EXCHANGE_RATES_APP_ID) {
+    throw new Error('OPEN_EXCHANGE_RATES_APP_ID is not set');
+  }
+
+  try {
+    const params = new URLSearchParams({
+      app_id: env.OPEN_EXCHANGE_RATES_APP_ID,
+    });
+    const data = await retryFetchParse<OpenExchangeRates>(
```

Comment:
Could connect Garmr, it has retry support and we can also fallback and trigger proper errors with it if this fails for any reason as 3party service.

Reply:
I think it might be a good idea as well.
Just another failsafe

Reply:
Added it